### DISCUSSION TITLE
Add update feature on Mark class

### DISCUSSION THREAD

**Discussion by aledeg** [maize-tech/laravel-markable#55] [Ideas]
# Context
I am using a custom `Mark` class to handle recently viewed items. This class is called through an event in some of my controllers. The table is properly handled as I can see items piling in there as soon as they are triggered.

# Encountered problems
When I trigger the controller for the same model multiple times, it's always the first one that is kept. It means I have no way to know when was the last time I visited a model.

My first though was to handle that with a `Recent::remove` followed by `Recent::add`. This works but that means that my ids will increase rapidly.

I then tried to handle that through `Recent::toggle` but I have the exact same problem.

# Current solution
I used a manual update to do such thing with something similar to the following:
```php
Recent::where('markable_id', $markable->getKey()
    ->where('markable_type', $markable->getMorphClass())
    ->where('user_id', Auth::id())
    ->update([
        'updated_at' => now(),
    ]);
```
This works but it's not very elegant.

# Feature request
I think this could be handled directly at the library level with some custom method on the `Mark` class. Something similar to `Mark::addOrUpdate($markable, $user)`. But `Mark::update($markable, $user)` would work fine too.

**Comment by riccardodallavia** [comment]
Dear @aledeg,

you could actually work with the `add` method which returns the Mark instance and call laravel's `touch` method.
This way, the `add` method itself already creates a mark if none exists, or returns the existing one, whereas the `touch` method will update the entity timestamps.

Here's a working example:

```php
$article = Article::firstOrFail();
$user = User::firstOrFail();

Like::add($article, $user)->touch();
```

Please let me know if that works!

**Reply by aledeg** [reply]
Thank you @riccardodallavia. It works perfectly!
